zk-token-sdk/src/zk_token_elgamal/pod/mod.rs

```diff
@@ -73,3 +82,27 @@ impl TryFrom<PodProofType> for ProofType {
 #[derive(Clone, Copy, Pod, Zeroable, PartialEq, Eq)]
 #[repr(transparent)]
 pub struct CompressedRistretto(pub [u8; 32]);
+
+macro_rules! impl_from_str {
+    (TYPE = $type:ident, BYTES_LEN = $bytes_len:expr, BASE64_LEN = $base64_len:expr) => {
+        impl std::str::FromStr for $type {
+            type Err = crate::zk_token_elgamal::pod::ParseError;
+
+            fn from_str(s: &str) -> Result<Self, Self::Err> {
+                if s.len() > $base64_len {
+                    return Err(ParseError::WrongSize);
```

Comment:
Sorry one last bit: it seems like these usages here still require implementations to import `ParseError`, could you fully qualify these with `Self::Err` or `crate::zk_token_elgamal::pod::ParseError` so the import isn't required?

Reply:
Oh yes I should have caught this before. I removed the need for the dependency!